# Bubble Data EDA
- 2020-05-10 21:45:38
- A first look at the bubble data (from Zhiheng)
- Structure in data directory:
    - bubnums.csv: timestamp of event, "by-hand" count of number of bubbles
    - trig (dir): images during trigger (TIMESTAMP_trig.bmp)
    - next (dir): images in frame after trigger (TIMESTAMP_next.bmp)
    - diff (dir): images subtracting next - trig (TIMESTAMP_diff.bmp)

# Imports

In [2]:
import os
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
datadir = "/home/ckampa/Coding/machine_learning/data/"
datadir = os.getcwd()

# Load bubnums.csv

In [3]:
df = pd.read_csv(datadir+"bubnums.csv", names=["timestamp", "N_bubs"])

In [4]:
df.head()

timestamp  N_bubs
0  20170623_3_0       0
1  20170623_3_1       0
2  20170623_3_2       0
3  20170623_3_3       0
4  20170623_3_4       0

In [5]:
df.describe()

N_bubs
count  24774.000000
mean       0.307823
std        0.488041
min        0.000000
25%        0.000000
50%        0.000000
75%        1.000000
max        4.000000

In [6]:
(df.N_bubs > 0).sum()

7333

In [7]:
(df.N_bubs > 0).sum() / len(df)

0.2959958020505368

In [8]:
len(df)

24774

- Should have 24,774 datafiles.

# Count files in image directories

In [4]:
len(os.listdir(datadir))

7

In [6]:
len(os.listdir(datadir+'/trig/'))

24537

In [8]:
len(os.listdir(datadir+'/next/'))

24536

In [9]:
len(os.listdir(datadir+'/diff/'))

24536